core/src/cluster_slots_service/cluster_slots.rs

```diff
@@ -200,17 +200,21 @@ impl ClusterSlots {
         &self,
         slot: Slot,
         repair_peers: &[ContactInfo],
-    ) -> Vec<(u64, usize)> {
-        self.lookup(slot)
-            .map(|slot_peers| {
-                let slot_peers = slot_peers.read().unwrap();
-                repair_peers
-                    .iter()
-                    .enumerate()
-                    .filter_map(|(i, ci)| Some((slot_peers.get(ci.pubkey())? + 1, i)))
-                    .collect()
-            })
-            .unwrap_or_default()
+    ) -> (Vec<u64>, Vec<usize>) {
+        let mut weights = Vec::with_capacity(repair_peers.len());
+        let mut indices = Vec::with_capacity(repair_peers.len());
+
+        let Some(slot_peers) = self.lookup(slot) else {
+            return (weights, indices);
+        };
```

Comment:
Do you know how often this should be the case? We could refactor to return `vec![]` (which doesn't allocate) in this case and only after this point do `Vec::with_capacity(repair_peers.len());` (now that we know we'll actually populate it)

Reply:
this should not happen at all under normal operation. i will change to not allocate unnecessarily though, just in case.

Reply:
Yes, I will be making several on repair. Stay tuned